ambry-router/src/test/java/com.github.ambry.router/GetOperationTest.java

```diff
@@ -0,0 +1,474 @@
+  private final MockClusterMap mockClusterMap;
+  private final MockServerLayout mockServerLayout;
+  private final int replicasCount;
+  private final AtomicReference<MockSelectorState> mockSelectorState = new AtomicReference<>();
+  private final ResponseHandler responseHandler;
+  private final MockNetworkClientFactory networkClientFactory;
+  private final NetworkClient networkClient;
+  private final MockTime time;
+  private final Map<Integer, GetOperation> correlationIdToGetOperation = new HashMap<>();
+  private final NonBlockingRouter router;
+  private final int BLOB_SIZE = 100;
+  private final int BLOB_USER_METADATA_SIZE = 10;
+  private final Random random = new Random();
+  private final String blobIdStr;
+  private final BlobProperties blobProperties;
+  private final byte[] userMetadata;
+  private final byte[] putContent;
+  private final GetTestRequestRegistrationCallbackImpl requestRegistrationCallback =
+      new GetTestRequestRegistrationCallbackImpl();
+  private final FutureResult<BlobInfo> operationFuture = new FutureResult<>();
+  private final AtomicReference<BlobInfo> operationResult = new AtomicReference<>(null);
+  private final AtomicReference<Exception> operationException = new AtomicReference<>(null);
+  private final Callback<BlobInfo> operationCallback = new Callback<BlobInfo>() {
+    @Override
+    public void onCompletion(BlobInfo result, Exception exception) {
+      operationResult.set(result);
+      operationException.set(exception);
+    }
+  };
+
+  private class GetTestRequestRegistrationCallbackImpl implements RequestRegistrationCallback<GetOperation> {
+    private List<RequestInfo> requestListToFill;
+
+    @Override
+    public void registerRequestToSend(GetOperation getOperation, RequestInfo requestInfo) {
+      requestListToFill.add(requestInfo);
+      correlationIdToGetOperation.put(((RequestOrResponse) requestInfo.getRequest()).getCorrelationId(), getOperation);
+    }
+  }
+
+  ;
+
+  public GetOperationTest()
+      throws Exception {
+    VerifiableProperties vprops = getNonBlockingRouterProperties();
+    routerConfig = new RouterConfig(vprops);
+    mockClusterMap = new MockClusterMap();
+    mockServerLayout = new MockServerLayout(mockClusterMap);
+    replicasCount = mockClusterMap.getWritablePartitionIds().get(0).getReplicaIds().size();
+    responseHandler = new ResponseHandler(mockClusterMap);
+    time = new MockTime();
+    networkClientFactory = new MockNetworkClientFactory(vprops, mockSelectorState, MAX_PORTS_PLAIN_TEXT, MAX_PORTS_SSL,
+        CHECKOUT_TIMEOUT_MS, mockServerLayout, time);
+    router = new NonBlockingRouter(new RouterConfig(vprops), new NonBlockingRouterMetrics(new MetricRegistry()),
+        networkClientFactory, new LoggingNotificationSystem(), mockClusterMap, time);
+    blobProperties = new BlobProperties(BLOB_SIZE, "serviceId", "memberId", "contentType", false, Utils.Infinite_Time);
+    userMetadata = new byte[BLOB_USER_METADATA_SIZE];
```

Comment:
I think it is fine to initialize it here, since these arrays are filled with random bytes and that is part of the initialization which cannot be done outside.